### PR TITLE
Update to new config API

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Refactored config feature to adapt to the new config API https://github.com/temporalio/tctl-kit/pull/23

## Why?

the new API is way more stable 

## Checklist

1. Closes 

2. How was this tested:

updated unit tests

3. Any docs updates needed?

### CODE REVIEW

File: cli/config.go [hunk 1]
```diff
@@ -23,108 +23,110 @@
...
 	}
 
-	fmt.Printf("%v: %v\n", color.Magenta(c, "%v", key), val)
+	env, key := envKey(fullKey)
+
+	if err := tctlConfig.SetEnvProperty(env, key, val); err != nil {
+		return fmt.Errorf("unable to set env property %v. %s", key, err)
+	}
 
+	fmt.Printf("%v: %v\n", key, val)
```
Comment (bergundy) [inline cli/config.go:103]: We should print the full key here to let user understand which env the key was set for.

```suggestion
	fmt.Printf("Set '%v' to: %v\n", fullKey, val)
```

File: cli/config.go [hunk 2]
```diff
@@ -23,108 +23,110 @@
...
-				Name:     "alias",
-				Usage:    "Alias for command",
-				Required: true,
-				Value:    "workflow list --output json",
-			},
-		},
-		Action: func(c *cli.Context) error {
-			return createAlias(c)
-		},
+func validateEnvKey(fullKey string) error {
```
Comment (bergundy) [inline cli/config.go:107]: Probably should validate that the first key is `env` if given a full path and that the last key is something that's supported by tctl.

Comment (feedmeapples) [reply cli/config.go:107]: added first key `env` valiadation

for the last key validation i guess will need to traverse through all of the commands and collect a list of flags to validate against

**Review by bergundy — APPROVED** [review]
Blind approve, sorry for not spending more time going over the code.

**Pull request merged**